src/ZoneServer/Scripting/AI/AiScript.Routines.cs

```diff
@@ -238,16 +260,18 @@ protected IEnumerable Follow(ICombatEntity followTarget, float minDistance = 50,
 					// recreate it on the other side, from the summoning
 					// script. All this could need a clean up.
 
-					if (!ZoneServer.Instance.Conf.World.BlueOrbFollowWarp)
+					if (!ZoneServer.Instance.Conf.World.BlueOrbFollowWarp && !ZoneServer.Instance.Conf.World.BlueOrbPetSystem)
 					{
 						while (true)
 							yield return this.Wait(10000);
 					}
 					else
 					{
 						if (this.Entity is Mob mob)
+						{
 							this.Entity.Map.RemoveMonster(mob);
-						yield break;
+							yield break;
```

Comment:
The yield break coming either way after the if was intentional, is there a reason for this change? Because now, if the entity were a player for example, it wouldn't break, and the following would continue, which we don't want if the target switched maps.

Reply:
Nope, that was a mistake, at one point I had more code there, it got messed up when I removed it.  I've set it back to the original now.